distribution/sender/routes/sender-activemq-publisher.xml

```diff
@@ -0,0 +1,149 @@
+                                    </spel>
+                                </script>
+                            </when>
+            
+                            <script>
+                                <spel>
+                                    #{body.metadata.setSourceIdentifier('{{db-sync.senderId}}')}
+                                    #{body.metadata.setDateSent(T(java.time.LocalDateTime).now())}
+                                    #{body.metadata.setOperation(getProperty('senderSyncMessage').operation)}
+                                    #{body.metadata.setRequestUuid(getProperty('senderSyncMessage').requestUuid)}
+                                    #{body.metadata.setMessageUuid(getProperty('senderSyncMessage').messageUuid)}
+                                    #{body.metadata.setSnapshot(getProperty('senderSyncMessage').snapshot)}
+                                </spel>
+                            </script>
+                            <setBody>
+                                <method beanType="org.openmrs.eip.component.utils.JsonUtils" method="marshall(${body})" />
+                            </setBody>
+            
+                            <log loggingLevel="DEBUG" message="Sync payload -> ${body}" />
+    
+                            <when>
+                                <simple>{{openmrs.eip.dbsync.encryption.enabled}} == true</simple>
+                                <log message="Encrypting entity payload.." />
+            
+                                <process ref="pgpEncryptService" />
+            
+                                <log message="Encrypted entity payload -> ${body}" loggingLevel="TRACE" />
+                            </when>
+    
+                            <log message="Sending entity to sync destination: {{camel.output.endpoint}}" />
+            
+                            <toD uri="{{camel.output.endpoint}}" />
+            
+                            <log message="Entity sent!" loggingLevel="DEBUG" />
+                    
+                            <script>
+                                <spel>
+                                    #{getProperty('senderSyncMessage').markAsSent()}
+                                </spel>
+                            </script>
+                        </when>
+                        <otherwise>
+                            <setProperty name="errorMessage">
+                                <constant>No entity found in the database matching identifier</constant>
+                            </setProperty>
+                            
+                            <log message="${exchangeProperty.errorMessage}: ${exchangeProperty.senderSyncMessage.identifier}" loggingLevel="ERROR" />
+                            <script>
+                                <spel>
+                                    #{getProperty('senderSyncMessage').markAsError(getProperty('errorMessage'))}
```

Comment:
**SenderSyncMessage** can't have error state, if something fails the applications should stop which why you set the error handler to shutdownErrorHandler

Reply:
Please don't change the logic, if no entity is found, you do nothing just like the old code

Reply:
Please fix this code to behave like before in master, you should send send the message back to the receiver with no entity payload, don't change the status to ERROR, infact remove the ERROR status from the status enum and remove the error_message field.

Reply:
I think that there is a misunderstand here. Lets go though each point:

_"SenderSyncMessage can't have error state, if something fails the applications should stop which why you set the error handler to shutdownErrorHandler"_
The only error that will be in this column for now is the one that was only logged, and that has a TODO to persist it in the DB (see [this comment](https://github.com/FriendsInGlobalHealth/openmrs-eip/pull/11#discussion_r907182671)). All other errors will remain lead to stop the application. 

_"Please don't change the logic, if no entity is found, you do nothing just like the old code"_
There is two scenarios where we can face this, which have different flows:
1. if we do not found the entity for Sync Requests from Central, we send an empty payload. This logic did not changed.

2. if we do not found the entity for all the other events, like create/update/delete, then we do not send an empty payload. In the previous version, we only logged this error and there was a TODO (see this 8667f83535be2c1dad29587224f4c4c5a147f988). And now, since we have a SenderSyncMessage register in the database, we cannot only log. For now, we mark it as ERROR to avoid reprocessing, and write the error in the column.
If this error information must not be there, we need to decide what to do with this SenderSyncMessages:
2.1. Mark with ERROR only, without error message
2.2. Simply log it like before, and remove the SenderSyncMessage to avoid reprocessing.

_"Please fix this code to behave like before in master, you should send send the message back to the receiver with no entity payload, don't change the status to ERROR, infact remove the ERROR status from the status enum and remove the error_message field."_
Please see the previous comment.

Reply:
When processing a regular non delete DB event(insert/update) i.e. one where operation property is not set to **r** and the entity is not found, it means there is nothing to send the receiver, just log the message and delete the sync message row.

Reply:
Implemented accordingly.